binary_search.rb

```diff
@@ -0,0 +1,33 @@
+# input = array siempre ordenado
+# input = target el entero que estamos buscando
+# output = el índice de la posición del target en el array.
+
+def binary_search(array, target)
+  return -1 if array.length == 0
+  array_length = array.length # 4
+  lo = 0
+  hi = array_length - 1 # 3
+  iterate = true
+  while true
+    middle_index = (lo + hi) / 2
```

Comment:
mid = low + ((high - low) / 2);

Es un error común en búsquedas binarias. Java tuve ese problema por décadas: http://googleresearch.blogspot.com/2006/06/extra-extra-read-all-about-it-nearly.html